src/cpp/fastdds/xtypes/dynamic_types/DynamicDataImpl.cpp

```diff
@@ -1537,6 +1542,22 @@ traits<DynamicType>::ref_type DynamicDataImpl::type() noexcept
     return type_;
 }
 
+traits<DynamicTypeImpl>::ref_type DynamicDataImpl::enclosing_type() noexcept
+{
+    return enclosing_type_;
+}
```

Comment:
There is already an `get_enclosing_type` function.

Reply:
However its purpose is different.